bundles/com.zeligsoft.domain.ngc.ccm.descriptorgeneration/template/mainTransformAxcioma.ext

```diff
@@ -847,26 +847,49 @@ create PlanConnectionDescription createSyncClientSideConnection(InstanceDeployme
 	let sourceSN = source.getScopedName() :
 	let facetSN = facetPart.getScopedName() :
 	let compInstance = plan.instance.selectFirst( i | i.name.first().matches(sourceSN)) :
-	this.name.add(sourceSN + "." + receptacle.name + "::" + idd.name.first() + ".srr_facet") ->
+	let receptacleConnectorType = receptacle.connectorType.name:

```

Comment:
The same block of code appears several times across 200-300 lines of code. This should be wrapped in a function so there is a single point of change.Two functions such getConnectorProvidesPortName and getConnectorUsesPortName would eliminate the repetition. 

The use of functions might also allow you get get away from the use of a collection for connectorPortName, when clearly, it always has a single element.